### ISSUE TITLE
Sortable creating a new instance each change detection pass

### ISSUE THREAD

**Issue by maximedupre** [telerik/kendo-angular#118]
<template let-i="index" let-activeFx="item" >
        <my-component></my-component>
    <template let-i="index" let-activeFx="item" >

A new `my-component` will be instantiated every time there is a change detection pass (even if `data` has not changed). It should behave like a normal loop instead.

**Comment by Xizario** [comment]
@maximedupre, yes you will get new instance every time the template it rendered. This is rather expected.
What do you mean by normal loop?

**Comment by maximedupre** [comment]
If data doesn't change, there is no reason for the components inside the loop to be re-instantiated. By normal loop, I'm talking about a standard `ngFor` loop.

    <template ngFor let-track [ngForOf]="tracks">
        <track [trackName]="track.trackName"></track>
    </template>

If `tracks` doesn't change, no other `TrackComponent` is instantiated, even if the template is rendered 50 times. And even if it does change (e.g. an element is added), only the affected elements will destroy or create a `TrackComponent`.

It makes no sense to have a new instance every time the template is rendered, as a template, even in a small application, can be rendered hundreds of times per second (every time there is a change detection pass).

**Comment by Xizario** [comment]
But it already uses ngFor and works as you describe. 
```
@Component(
  {
    selector: 'my-component',
    template: `
    something here
  `
  }
)
export class MyComponent implements OnInit {
  public ngOnInit(): void {
    console.log("init");
  }
}

@Component({
  selector: 'my-app',
  template: `
<div>
  <kendo-sortable [data]="someData">
    <template let-item="item">      
        {{item}}       
        <my-component></my-component>      
    </template>
  </kendo-sortable>
</div>
  `
})
```
http://www.screencast.com/t/n6JYttEvo
As you can see the Init is fired only for the elements that are actually changed, not for all of them.

**Comment by maximedupre** [comment]
@Xizario You are right. The problem occurs when there is an element with a `ngIf` directive inside the Sortable template. I have not been able to reproduce the problem outside the `kendo-sortable` component in my application, but neither have I been able to reproduce the problem inside the `kendo-sortable` component with a basic plunker.

    <kendo-sortable [data]="activeFxs" [zone]="'activeFxsZone'" [acceptZones]="['fxsListZone']">
        <template let-item="item">
            <div *ngIf="true === false">Test</div>
        </template>
    </kendo-sortable>

Basically, whatever condition I put in the `ngIf` doesn't matter, the div element is always rendered. It seems like the `ngIf` condition isn't parsed correctly. The angular comment that reflects the template binding confirms that:

    

Instead, `"ng-reflect-ng-if": "[object Object]"` should be `"ng-reflect-ng-if": "false"`. I have no idea why the condition is evaluated to an object.

**Comment by Xizario** [comment]
I tried as well to reproduce it in basic page, but evaluates to "ng-reflect-ng-if": "false".  In the example below see the lines above "some text", and the "Test" div is not visible.

```
<kendo-sortable [data] = "palettes">
    <template let-palette="item">
            <div *ngIf="true === false">Test</div>
            some text
    </template>
</kendo-sortable>
```
```
<div ng-reflect-ng-class="" ng-reflect-ng-style="[object Object]" ng-reflect-draggable="true" draggable="true" aria-disabled="false" aria-grabbed="false" aria-dropeffect="move" data-sortable-index="0" data-sortable-id="0">
      <!--template bindings={
  "ng-reflect-ng-if": "[object Object]",
  "ng-reflect-ng-outlet-context": "[object Object]",
  "ng-reflect-ng-template-outlet": "[object Object]"
}-->
      
<!--template bindings={
  "ng-reflect-ng-if": "false"
}-->
some text
    
<!--template bindings={
  "ng-reflect-ng-if": "false"
}-->
</div>
```
If we look at the output:
In the first binding it checks is a template specified (the object is TemplateRef)
Then it comes your template.
Third one is if there is no template specified, which is false.

It could be improved, by placing the "is template specified" check outside the ngfor that renders the items.
But this still does not affect your case, until we find out why it behaves differently in your application and in the sample page.

**Comment by maximedupre** [comment]
Then my problem is even more weird. It seems like the `ngIf` is totally ignore in my case.

For this template:

    <kendo-sortable [data]="activeFxs" [zone]="'activeFxsZone'" [acceptZones]="['fxsListZone']">
        <template let-item="item">
            <div *ngIf="true === false">Test</div>
        </template>
    </kendo-sortable>

I get the HTML:

    <div ng-reflect-ng-style="[object Object]" ng-reflect-draggable="true" draggable="true" aria-disabled="false" aria-grabbed="false" aria-dropeffect="move" data-sortable-index="0" data-sortable-id="3">
        

        <div _ngcontent-myx-43="">Test</div>

        
    </div>


As you can see, there is no template binding comment reflecting the condition of my div. In your example you have and I am missing:

    

Which is why I though 

    

was the corresponding binding to my condition.
